### PR TITLE
Xcloud

### PR DESCRIPTION
**Pull request by Hanjun-Dai**
stablize jobs
- xm train/load
- fix grad explosion issue

### CODE REVIEW

File: crossbeam/xm_train.py [hunk 1]
```diff
@@ -48,8 +48,8 @@ def main(argv) -> None:
   uname = getpass.getuser()
 
   with xm_abc.create_experiment(experiment_title=_EXP_NAME.value) as experiment:
-    job_requirements = xm.JobRequirements(ram=26 * FLAGS.num_gpus * xm.GiB,
-                                          cpu=7 * FLAGS.num_gpus,
+    job_requirements = xm.JobRequirements(ram=8 * FLAGS.num_gpus * xm.GiB,
```
Comment (kensens) [inline crossbeam/xm_train.py:51]: Is it needed to reduce our resource consumption here? We can request the amount we were using before: https://g3doc.corp.google.com/company/teams/brain-frameworks/xcloud/guide.md?cl=head

File: crossbeam/experiment/train_eval.py [hunk 2]
```diff
@@ -191,7 +191,8 @@ def _gather_eval_info(rank, device, local_acc, local_num):
 
 
 def train_eval_loop(args, device, model, train_files, eval_tasks,
-                    task_gen, trace_gen):
+                    task_gen, trace_gen, checkpoint):
+  random.shuffle(train_files)
```
Comment (kensens) [inline crossbeam/experiment/train_eval.py]: I think we shouldn't shuffle train_files so that training runs are more consistent/reproducible

File: crossbeam/experiment/train_eval.py [hunk 3]
```diff
@@ -218,6 +219,13 @@ def local_task_gen(domain):
       os.makedirs(log_folder)
     log_writer = tensorboard.SummaryWriter(log_folder)
   model = model.to(device)
+  optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+  if checkpoint is not None:
+    model.load_state_dict(checkpoint['model'])
+    optimizer.load_state_dict(checkpoint['optimizer'])
+    starting_step = checkpoint['step']
```
Comment (kensens) [inline crossbeam/experiment/train_eval.py:220]: Can we skip `starting_step` batches from the training dataset, so that we don't train on the same data again upon restart? I think this is just:

```
for _ in range(starting_step):
  next(train_gen)
```

**Review by kensens — APPROVED** [review]
Thanks Hanjun! I left a few comments
